test/end2end_test.go

```diff
@@ -7766,3 +7766,67 @@ func (cvd *credentialsVerifyDeadline) Clone() credentials.TransportCredentials {
 func (cvd *credentialsVerifyDeadline) OverrideServerName(s string) error {
 	return nil
 }
+
+func unaryInterceptorVerifyConn(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	conn := transport.GetConnection(ctx)
+	if conn == nil {
+		return nil, status.Error(codes.NotFound, "connection was not in context")
+	}
+	return nil, status.Error(codes.OK, "")
+}
+
+// TestUnaryServerInterceptorGetsConnection tests whether the accepted conn on
+// the server gets to any unary interceptors on the server side.
+func (s) TestUnaryServerInterceptorGetsConnection(t *testing.T) {
+	ss := &stubserver.StubServer{}
+	if err := ss.Start([]grpc.ServerOption{grpc.UnaryInterceptor(unaryInterceptorVerifyConn)}); err != nil {
+		t.Fatalf("Error starting endpoint server: %v", err)
+	}
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	if _, err := ss.Client.EmptyCall(ctx, &testpb.Empty{}); status.Code(err) != codes.OK {
+		t.Fatalf("%v.EmptyCall(_, _) = _, %v, want _, error code %s", ss.Client, err, codes.OK)
+	}
+}
+
+func streamingInterceptorVerifyConn(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	conn := transport.GetConnection(ss.Context())
+	if conn == nil {
+		return status.Error(codes.NotFound, "connection was not in context")
+	}
+	return status.Error(codes.OK, "")
+}
+
+// TestStreamingServerInterceptorGetsConnection tests whether the accepted conn on
+// the server gets to any streaming interceptors on the server side.
+func (s) TestStreamingServerInterceptorGetsConnection(t *testing.T) {
+	ss := &stubserver.StubServer{}
+	if err := ss.Start([]grpc.ServerOption{grpc.StreamInterceptor(streamingInterceptorVerifyConn)}); err != nil {
+		t.Fatalf("Error starting endpoint server: %v", err)
+	}
+	defer ss.Stop()
+
+	respParam := []*testpb.ResponseParameters{
+		{
+			Size: int32(1),
+		},
+	}
+	payload, err := newPayload(testpb.PayloadType_COMPRESSABLE, int32(1))
+	if err != nil {
+		t.Fatal(err)
+	}
+	req := &testpb.StreamingOutputCallRequest{
+		ResponseType:       testpb.PayloadType_COMPRESSABLE,
+		ResponseParameters: respParam,
+		Payload:            payload,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	_, err = ss.Client.StreamingOutputCall(ctx, req)
+	if status.Code(err) != codes.OK {
+		t.Fatalf("%v.StreamingOutputCall(_) = _, %v, want _, error code %s", ss.Client, err, codes.OK)
+	}
```

Comment:
This isn't how the streaming call would return the error if there was one (try changing your interceptor to always return an error and see).  You should do a `stream.Recv` on the returned stream object to get the error instead.  (You should also ensure the error returned by `StreamingOutputCall` is `nil` always.)

Reply:
I think this is correct. The interceptor intercepts before stream creation, and returns an error (see my interceptor). Thus, this is the error from the interceptor, and not the error from the stream.